src/goCheck.ts

```diff
@@ -106,6 +106,43 @@ function runTool(args: string[], cwd: string, severity: string, useStdErr: boole
 	});
 }
 
+/**
+ * Check if the given path is a directory.
+ * @param filePath Path to the potential directory
+ */
+function isDirectory(filePath: string): boolean {
+	try {
+		return fs.statSync(filePath).isDirectory();
+	} catch (e) {
+		return false;
+	}
+}
+
+/**
+ * List directories (including the given dirPath) which contain files with the '_test.go' suffix.
+ * This ignores hidden directories and their sub-directories.
+ * @param dirPath Directory in which to start searching
+ */
+function findTestDirectories(dirPath: string): string[] {
```

Comment:
I've had the exact same thought, but I've come to the conclusion that it doesn't matter. If you set it to build the entire workspace, you should expect that to happen. In case somebody uses a big workspace, they can set it to build only the current package in the workspace-local settings.

Additionally, even though the directory traversal uses synchronous functions, it is way faster than the lint/vet/etc jobs. It appears that the extension accumulates the results of all "check"-Promises anyway, so delaying some for a couple milliseconds doesn't really hurt.

I understand and agree that this is not a particularly smart solution, but it should work for most use cases.

> Also, I'd much rather prefer a better way to compile test files from Go itself.

Not sure, if that is going to happen. The only smart alternative is to run `go test -run ! ./...`, which will check the test files, but won't run any  since they don't match the pattern. Could you work with that?